bridge/opentracing/wrapper.go

```diff
@@ -24,21 +25,62 @@ import (
 // WrapperTracerProvider is an OpenTelemetry TracerProvider that wraps an
 // OpenTracing Tracer.
 type WrapperTracerProvider struct {
-	wTracer *WrapperTracer
+	getWrappedTracer func(name string, opts ...trace.TracerOption) *WrapperTracer
 }
 
 var _ trace.TracerProvider = (*WrapperTracerProvider)(nil)
 
-// Tracer returns the WrapperTracer associated with the WrapperTracerProvider.
-func (p *WrapperTracerProvider) Tracer(_ string, _ ...trace.TracerOption) trace.Tracer {
-	return p.wTracer
+// Tracer creates a WrapperTracer associated with the WrapperTracerProvider.
+func (p *WrapperTracerProvider) Tracer(name string, opts ...trace.TracerOption) trace.Tracer {
+	return p.getWrappedTracer(name, opts...)
 }
 
+// Deprecated: Use NewTracerProvider(...) instead.
+//
 // NewWrappedTracerProvider creates a new trace provider that creates a single
-// instance of WrapperTracer that wraps OpenTelemetry tracer.
+// instance of WrapperTracer that wraps OpenTelemetry tracer, and always returns
+// it unmodified from Tracer().
 func NewWrappedTracerProvider(bridge *BridgeTracer, tracer trace.Tracer) *WrapperTracerProvider {
+	wTracer := NewWrapperTracer(bridge, tracer)
 	return &WrapperTracerProvider{
-		wTracer: NewWrapperTracer(bridge, tracer),
+		getWrappedTracer: func(_ string, _ ...trace.TracerOption) *WrapperTracer {
+			return wTracer
+		},
+	}
+}
+
+type wrappedTracerKey struct {
+	name    string
+	version string
+}
+
+// NewTracerProvider creates a new trace provider that creates new instances of
+// WrapperTracer that wraps OpenTelemetry tracer for each call to Tracer().
+func NewTracerProvider(bridge *BridgeTracer, provider trace.TracerProvider) *WrapperTracerProvider {
```

Comment:
If we are deprecating in favor of clarifying and extending ability, I think this should return a new `TracerProvider` type added to this package. That type would implement the "lookup tracer" functionality and would distinguishable from the `WrappedTracerProvider` functionality.

Reply:
I've rewritten the new funcitonality as a new type under `provider.go`!